koku/common/__init__.py

```diff
@@ -0,0 +1,63 @@
+import functools
+import logging
+import random
+import time
+import typing as t
+
+from api.common import log_json
+from common.util import extract_context_from_sql_params
+
+LOG = logging.getLogger(__name__)
+
+
+def retry(
+    original_callable: t.Callable = None,
+    *,
+    retries: int = 3,
```

Comment:
Most of the places that redundant retry code can be removed `settings.HIVE_PARTITION_DELETE_RETRIES` is used for the number of retries. Setting that as a default here seems sensible.
```suggestion
    retries: int = settings.HIVE_PARTITION_DELETE_RETRIES,
```